src/network/networkd-link.c

```diff
@@ -261,6 +261,20 @@ static int link_enable_ipv6(Link *link) {
         return 0;
 }
 
+static bool link_is_bridge_port(Link *link) {
+
+        assert(link);
+
+        if (link->flags & IFF_LOOPBACK)
+                return false;
+
+        if (link->network)
+                if (link->network->bridge)
+                        return true;
```

Comment:
> if a link is already a bridge port and it's not in the configuration

In that case your patch will do nothing? Also if it's "not in the configuration", why would networkd handle it *at all*?

I don't mind testing it, even though I think we can see what will happen with the code. But can you at least be more specific on "not in the configuration"? You mean no .network file at all for the slave or somehow an inapproriate .network file with arbitrary options but not `Bridge=`?